node/overseer/src/lib.rs

```diff
@@ -1514,7 +1514,9 @@ where
 			update.activated.push((hash, span));
 		}
 
-		self.broadcast_signal(OverseerSignal::ActiveLeaves(update)).await?;
+		if !update.is_empty() {
+			self.broadcast_signal(OverseerSignal::ActiveLeaves(update)).await?;
```

Comment:
I would prefer for the overseer at least to notify subsystems of finality. We shouldn't delay these notifications by seconds as subsystems may want to act on them. The goal of this PR can be handled completely on the network bridge